### ISSUE TITLE
Tracking issues for spurious failures related to IPv6 problems in Travis CI's new 2017-Q4 image

### ISSUE THREAD

**Issue by kennytm** [rust-lang/rust#47002]
Upstream issue: travis-ci/travis-ci#8891.

## 1. ~~Network tests related to IPv6 are failing~~

Error first happens in #46692. Fixed in #44694 by using the older Travis image instead. 

## 2. ~~Docker failed to start due to "Address already in use"~~

The old image is reverted in #46924. It was still going well for ~5.5 hours (2 more PRs) after the PR is merged, but when bors try #46941, the jobs all immediately fail with

```
...
[00:01:45] Sending build context to Docker daemon  499.2kB
[00:01:45] Step 1/10 : FROM ubuntu:16.04
[00:01:45]  ---> 00fd29ccc6f1
[00:01:45] Step 2/10 : RUN apt-get update && apt-get install -y --no-install-recommends   clang   make   file   curl   ca-certificates   python2.7   git   cmake   sudo   bzip2   xz-utils   wget   libssl-dev   pkg-config
[00:01:45]  ---> Using cache
[00:01:45]  ---> f82100b9eb89
[00:01:45] Step 3/10 : COPY scripts/freebsd-toolchain.sh /tmp/
[00:01:45]  ---> Using cache
[00:01:45]  ---> 41afc339a9b0
[00:01:45] Step 4/10 : RUN /tmp/freebsd-toolchain.sh i686
[00:01:45]  ---> Running in 66ec201d700e
[00:01:45] Address already in use
[00:01:45] The command has failed after 5 attempts.
```

The failure is extremely quick and rapidly burnt out the entire tree. This has been fixed by #47000.

**Comment by kennytm** [comment]
Timeline of the second bug:

1. 2017-12-23T14:12:10Z — #46924 rolled up into #46967
2. 2017-12-23T15:17:33Z — It was noted that the roll up [failed in Travis](https://travis-ci.org/rust-lang/rust/builds/320656619) with "Address already in use" while building docker image. #46924 is backed out from the rollup.
3. 2017-12-25T10:07:15Z — #46924 has been merged.
4. (#46949 and #46971 are merged successfully afterwards).
5. 2017-12-25T15:46:13Z — #46941 experienced the beginning of spurious failure.
6. 2017-12-25T18:37:43Z — Error noted, suspecting to be caused by the IPv6 address. Fix attempt filed at #47000.

**Comment by kennytm** [comment]
Since we are not seeing any more spurious errors related to this image for over 2 months, I consider the image is now stable and am closing this issue.
